fix(components): Make modal focus when opened

voodoods opened the pull request:
## Motivations

Improve component behaviour and contribute to hacktoberfest by providing a fix for https://github.com/GetJobber/atlantis/issues/385

### Fixed

- Fix focus on modal content on mount

### Security

- fixed React vulnerability by making conditional for `catchKeyboardEvent` hook part of the hook itself 

This follows the [Only Call Hooks on Top Level](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level) rule as stated by the React docs.

### Testing

- added unit test for checking focus on open modal
- manual test: open modal in playground (`yarn start` --> http://localhost:3333/components/modal) and press tab key


### Misc

- Please don't forget to add "hacktoberfest" and / or "hacktoberfest-accepted" labels to this pull request to enable [verification](https://hacktoberfest.digitalocean.com/faq). Thank you :)

eddysims commented:
@voodoods Looking good so far! A couple things:

1. `Escape` no longer closes the `Modal`
2. I believe that when the `Modal` is open we do not want the focus to be able to leave the modal. @seanhealy or @chris-at-jobber can either of you confirm this?

chris-at-jobber commented:
@eddysims For #2, that would be desirable and if we can solve for it in this PR that's fine, but I feel like it may fall outside of the scope of the initial improvement. One suggested solution I have seen is [the `inert` polyfill](https://github.com/GoogleChrome/inert-polyfill). We have a similar issue with [Lightbox](https://atlantis.getjobber.com/components/light-box) so I might suggest that we handle this separately.

@voodoods I've tested this with a screenreader as well and it has the desired effect of reading out the title, close button, content and actions. Other than Eddy's request to maintain the `Esc` behaviour this looks good to me!

voodoods commented:
@eddysims @chris-at-jobber Thanks for your feedback!

- [x] fixed `Escape` keyboard event
- [x] moved new hook to separate function call so to improve readability a little more

Hope I didn't miss anything else. Please have another look.

chris-at-jobber commented:
Behaviour looks good to me 👍 

@eddysims I'll leave review on the code to you
